### PR TITLE
[CS] Pass SelectedOverload in a couple of places

### PR DESCRIPTION
**Pull request by hamishknight**
Clean up a couple of members in CSApply by passing the `SelectedOverload` instead of individual parameters for things like the function ref kind, choice, and opened type.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test

**Review by xedin — APPROVED** [review]

File: lib/Sema/CSApply.cpp [hunk 1]
```diff
@@ -424,7 +422,7 @@ namespace {
                 // FIXME: the hop through 'getDecl()' is because
                 // SpecializedProtocolConformance doesn't substitute into
                 // witnesses' ConcreteDeclRefs.
-                Type expectedFnType = simplifiedFnType->getResult();
+                Type expectedFnType = simplifyType(overload.openedType);
```
Comment (CodaFi) [inline lib/Sema/CSApply.cpp]: I think you need the full opened type here.

Comment (hamishknight) [reply lib/Sema/CSApply.cpp]: @CodaFi Hmm, shouldn't the result of the full opened type be the opened type in this case?

Comment (CodaFi) [reply lib/Sema/CSApply.cpp]: Let’s add an assertion to that effect.

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

**Comment by hamishknight** [comment]
@swift-ci please test

File: lib/Sema/CSApply.cpp (hunk 1 above)
Comment (hamishknight) [reply lib/Sema/CSApply.cpp]: @CodaFi Sure thing! How does the updated code look?

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

**Comment by hamishknight** [comment]
@swift-ci please test source compatibility

File: lib/Sema/CSApply.cpp (hunk 1 above)
Comment (CodaFi) [reply lib/Sema/CSApply.cpp]: Great, thanks!

**Review by CodaFi — APPROVED** [review]

**Pull request merged**